### DISCUSSION TITLE
Guidance or docs on how to use existing web components libraries?

### DISCUSSION THREAD

**Discussion by leolorenzoluis** [fable-compiler/Fable.Lit#59] [Q&A] [answered]
If I want to use Google Charts web components, how do I integrate it with Fable? Do I just use importDynamic and also create the bindings for it?

**Comment by alfonsogarciacaro** [comment, marked as answer]
You just need to register them and then you can use them normally in the templates. It depends if the library requires you to call a `register` function explicitly or not. For example, Fluent UI requires it but Shoelace does not.

Jordan Marr does use `importDynamic` here: https://github.com/JordanMarr/fable-lit-fullstack-template/blob/ebb39776725190bea6915d82b30d62889bd7ac6b/Template/WebLit/src/Registrations.fs

But if you want to import things statically, `importSideEffects` should work too (this also allows you to include the components in your bundle). This example is for Solid, but it should work the same in Lit:

1. [Import the component](https://github.com/alfonsogarciacaro/Feliz.Solid/blob/7ebc7be4a12183333c00b9ae19b60bb31d2a8271/src/App/Components.fs#L496)
2. [Use it in a template](https://github.com/alfonsogarciacaro/Feliz.Solid/blob/7ebc7be4a12183333c00b9ae19b60bb31d2a8271/src/App/Components.fs#L553-L555)
